lib/lintSource.mjs

```diff
@@ -151,3 +154,29 @@ function createEmptyPostcssResult(filePath) {
 		warn: () => {},
 	};
 }
+
+/**
+ * even though stylelint-disable comments cannot be inserted inside a declaration or a selector list,
+ * new lines cannot be disregarded because FixerData['source'] will eventually be exposed
+ * i.e. it must be accurate
+ * @see stylelint/stylelint#7192
+ * @param {PostcssResult} result
+ */
+function applyFixes(result) {
+	const {
+		disabledRanges: { all },
+		fixersData,
+	} = result.stylelint;
+	const array = Object.values(fixersData).flat();
+	/** @type { Array<[number, number]> | undefined } */
+	const ranges = all?.map(({ start, end }) => [Number(start), Number(end)]);
+
+	array.forEach(({ source, callback, args, unfixable }) => {
+		const isInRange = (/** @type {[number, number]} */ [start, end]) =>
+			source?.start?.line && source.start.line >= start && source.start.line <= end;
+
+		if (!ranges || !ranges.some(isInRange)) {
+			!unfixable && callback(args);
```

Comment:
note

We don't bail out on the fixer addition because we will eventually have to expose the fix data regardless of the fixability.
i.e. it's simpler to do it that way than having different paths